src/services/MapService.ts

```diff
@@ -222,13 +227,93 @@ const getMapDataFromPlantingSite = (plantingSite: PlantingSite): MapData => {
   };
 };
 
+/**
+ * Extract Planting Site, Zones, Subzones, and Plots from an ObservationResult
+ */
+const getMapDataFromObservation = (observation: ObservationResults): MapData => {
```

Comment:
since we are using redux for observations.. perhaps leverage redux selectors to fetch this view of data, maybe create a new observationMapDataSelectors under redux/features/observations?

Reply:
Thinking about this more.. my suggestion doesn't make sense here for the current architecture. The selected data is already passed into the component. Please ignore.
Only way is to rearchitect and have the map component trigger selection of data from the store.. not sure there's much of a win here.

Reply:
Good point, I'll take a look at that